### PR TITLE
added default db params to dev file (fix)

### PR DESCRIPTION
**Pull request by arinkulshi-skylight**
## Description
Docker is not connecting to DB anymore. db params are updated to ensure db connection works. 

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [x] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

**Comment by arinkulshi-skylight** [comment]
> ```
> % docker compose -f dev-env.yaml up
> invalid interpolation format for services.db.environment.POSTGRES_DB.
> You may need to escape any $ with another $.
> ${postgres_db_name:reportvision}
> ```
> 
> This seems to fix it. I believe it uses bash parameter expansion? https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
> 
> ```diff
> diff --git a/dev-env.yaml b/dev-env.yaml
> index 14b607e..c384393 100644
> --- a/dev-env.yaml
> +++ b/dev-env.yaml
> @@ -31,12 +31,12 @@ services:
>      ports:
>        - "5432:5432"
>      environment:
> -      POSTGRES_DB: ${postgres_db_name:reportvision}
> -      POSTGRES_HOST: ${postgres_fqdn:localhost}
> -      POSTGRES_USER: ${postgres_user:postgres}
> -      POSTGRES_PASSWORD: ${postgres_password:super_secret_password}
> -      DB_PORT: ${postgres_port:5432}
> -      SSL_MODE: ${postgres_sslmode:disable}
> +      POSTGRES_DB: ${postgres_db_name:-reportvision}
> +      POSTGRES_HOST: ${postgres_fqdn:-localhost}
> +      POSTGRES_USER: ${postgres_user:-postgres}
> +      POSTGRES_PASSWORD: ${postgres_password:-super_secret_password}
> +      DB_PORT: ${postgres_port:-5432}
> +      SSL_MODE: ${postgres_sslmode:-disable}
>    api:
>      build: 
>        context: ./backend
> ```

You are right looks like I was using the springboot synthax here. I have made those changes.

File: backend/src/main/resources/application.yaml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 spring:
   datasource:
-    url: ${POSTGRES_HOST:postgres_fqdn}?sslmode=require
-    username: ${DB_USERNAME:postgres_user}
-    password:  ${POSTGRES_USER:postgres_password}
-    name: ${POSTGRES_DB:postgres_db_name}
+    url: jdbc:postgresql://${POSTGRES_HOST:localhost}:${DB_PORT:5432}/${POSTGRES_DB:reportvision}?sslmode=${SSL_MODE:disable}
```
Comment (marycrawford) [inline backend/src/main/resources/application.yaml:3]: Do we want to `enable` ssl_mode to enforce SSL encryption for this connection?

Comment (arinkulshi-skylight) [reply backend/src/main/resources/application.yaml:3]: I was thinking we can pass SSL_MODE as a argument(enabled,disabled) but kept it disabled by default for connecting to local host. Would this work?

**Comment by derekadombek** [comment]
In this PR we might want to add `ENTRYPOINT [ "./gradlew", "bootRun", "--continuous", "--args=--server.port=8081" ]` to the Dockerfile because right now that command is only being used in docker-compose.

**Comment by arinkulshi-skylight** [comment]
> In this PR we might want to add `ENTRYPOINT [ "./gradlew", "bootRun", "--continuous", "--args=--server.port=8081" ]` to the Dockerfile because right now that command is only being used in docker-compose.

I added it to the dev-dockerfile as well. Now it is in both places. Let me know if this makes sense or if want me to also remove it from the docker compose file.

**Comment by derekadombek** [comment]
> > In this PR we might want to add `ENTRYPOINT [ "./gradlew", "bootRun", "--continuous", "--args=--server.port=8081" ]` to the Dockerfile because right now that command is only being used in docker-compose.
> 
> I added it to the dev-dockerfile as well. Now it is in both places. Let me know if this makes sense or if want me to also remove it from the docker compose file.

I might have to change it more in the Dockerfile later on once more connections are in place but yeah its better to have the command in the dockerfile itself since not all systems use docker-compose commands.

File: dev-env.yaml [hunk 2]
```diff
@@ -31,11 +31,12 @@ services:
     ports:
       - "5432:5432"
     environment:
-      POSTGRES_DB: ${postgres_db_name}
-      POSTGRES_HOST: ${postgres_fqdn}
-      POSTGRES_USER:  ${postgres_user}
-      POSTGRES_PASSWORD: ${postgres_password}
-      sslmode: require
+      POSTGRES_DB: ${postgres_db_name:-reportvision}
```
Comment (marycrawford) [inline dev-env.yaml:34]: Does it matter if the `dev-env.yaml` is using `:-name` and the `application.yaml` file is not?

File: backend/src/main/resources/application.yaml (hunk 1 above)
Comment (marycrawford) [reply backend/src/main/resources/application.yaml]: It looks like @jonchang and @derekadombek may have tested this and it works as disabled. `SSL_MODE` enabled adds additional security.  However, I will defer to the majority decision of the team on whether it should be enabled at this time or before we hand it off.

**Review by derekadombek — APPROVED** [review]

File: backend/src/main/resources/application.yaml (hunk 1 above)
Comment (arinkulshi-skylight) [reply backend/src/main/resources/application.yaml]: ok we can set it as enabled by default then.

File: dev-env.yaml (hunk 2 above)
Comment (arinkulshi-skylight) [reply dev-env.yaml:34]: It seems one is springboot synthax and the other is bash synthax. app.yaml is being used locally so it should work as is.

**Pull request merged**